WordPress/gutenberg
# [RNMobile] Upgrade React Native to version 0.71.15

**Related PRs:**
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/6522
* https://github.com/wordpress-mobile/WordPress-Android/pull/19964
* https://github.com/wordpress-mobile/WordPress-iOS/pull/22410



## What?


Upgrade React Native to version 0.71.15.

## Why?


* To provide a long-term solution to the workaround introduced in https://github.com/WordPress/gutenberg/pull/57658.
* [Enable upgrading Android SDK to version 14](https://github.com/wordpress-mobile/gutenberg-mobile/issues/6533).

## How?


* Follow [the upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.71.11&to=0.71.15).
* Update `Podfile.lock` by running `npm run native preios`.
* Update `react-native` patch file.

> [!WARNING]
> Usually, in React Native upgrades we also update the version in [`react-native-libraries-publisher`](https://github.com/wordpress-mobile/react-native-libraries-publisher) to generate new Android binaries. However, since this upgrade is for a patch version, we assume all third-party libraries support any patch version of `0.71`. For this reason, this step has been skipped.

### Address build failure related to `RCT-Folly`

This is related to the issue shared in https://github.com/WordPress/gutenberg/pull/57667#issuecomment-1882953701. I investigated the problem further and found a workaround.

For some reason, `RCT-Folly` [redefines `clockid_t`](https://github.com/facebook/folly/blob/4baba28200d7446c870e96f3cdbeb492f54625d0/folly/portability/Time.h#L33), however, this seems not needed in newer iOS SDK versions. This was been fixed in `RCT-Folly` version (`v2022.08.29.00` - https://github.com/facebook/folly/commit/4a2410fae65afb85e1fec6d922005054b05de59f) but current React Native version points to an older version ([`2021.07.22.00` - reference](https://github.com/facebook/react-native/blob/5761ad6437b29dc559fce0914f2a8e71d02d25b3/third-party-podspecs/RCT-Folly.podspec#L6)). The version has been bumped in React Native version 0.73.0 ([`2022.05.16.00` - reference](https://github.com/facebook/react-native/blob/1eb4bf0d3af1fe267f5de6bf4bbe1e65d41fea1b/packages/react-native/third-party-podspecs/RCT-Folly.podspec#L6)), so we'd need to keep the patch until we upgrade.

Regarding the patch, this is the part we need from `react_native_pods` script ([reference](https://github.com/facebook/react-native/blob/5761ad6437b29dc559fce0914f2a8e71d02d25b3/scripts/react_native_pods.rb#L359-L364)) that removes the condition related to checking the minimum version required ([reference](https://github.com/facebook/folly/blob/4baba28200d7446c870e96f3cdbeb492f54625d0/folly/portability/Time.h#L33)):

## Testing Instructions




> [!NOTE]
> It's recommended to use the following installable builds for testing:
> * [Android](https://github.com/wordpress-mobile/WordPress-Android/pull/19964#issuecomment-1895427275)
> * [iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/22410#issuecomment-1895420518)

Nothing in particular, it should be enough by smoke testing the editor.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A
